ios/RN/RNCamera.m

```diff
@@ -522,7 +531,9 @@ - (void)record:(NSDictionary *)options resolve:(RCTPromiseResolveBlock)resolve r
         }
     }
 
-    [self updateSessionAudioIsMuted:[options[@"mute"] boolValue]];
+    if ([options valueForKey:@"mute"] != nil) {
```

Comment:
I also fixed this regression so it prevents the flickering if no mute property is present in `options["mute"]`. Nevertheless we should implement `defaultMuted` asap.